storage/watchable_store.go

```diff
@@ -308,6 +310,13 @@ func (s *watchableStore) syncWatchers() {
 		}
 		keyToUnsynced[k][w] = struct{}{}
 	}
+	prefixes := make([]string, len(keyToUnsynced))
+	i := 0
+	for k := range keyToUnsynced {
+		prefixes[i] = k
+		i++
+	}
+	sort.Strings(prefixes)
```

Comment:
`syncWatchers` is getting unwieldy; this bit and the loop to get the list of all events look ready to be refactored to a new function `eventsFromKVs(ks, vs) []storagepb.Event`